packages/vitest-pool-workers/src/pool/config.ts

```diff
@@ -221,6 +221,7 @@ async function parseCustomPoolOptions(
 		options.miniflare.assets = {
 			...options.miniflare.assets,
 			routingConfig: {
+				...options.miniflare.assets.routingConfig,
 				has_user_worker: Boolean(options.main),
 			},
 		};
```

Comment:
Nit: Would it be simpler to modify the config directly?, e.g.

```suggestion
		options.miniflare.assets.routingConfig ??= {};
		options.miniflare.assets.routingConfig.has_user_worker = Boolean(options.main);
```

Reply:
6a110217c546b6033b5ad939cb48a69c2bcd104e